nautobot/core/apps/__init__.py

```diff
@@ -76,6 +78,16 @@ def register_menu_items(tab_list):
 
                 group_perms = set()
                 for item in group.items:
+
+                    # Instead of passing the reverse url strings, we pass in the url itself initialized with args and kwargs.
+                    try:
+                        item.link = reverse(item.link, args=item.args, kwargs=item.kwargs)
+                    except NoReverseMatch as e:
+                        logger = logging.getLogger("nautobot.core.apps")
```

Comment:
Shouldn't be needed, we already have `logger` as a module global variable?

Reply:
Should I just use the default error message as in `logger.debug(f"{e}")` in that except block?

Reply:
Is the standard `NoReverseMatch` exception text at all helpful to the user? If so then I'd vote for `logger.debug("%s", e)`

Reply:
`Reverse for 'manufactur' not found. 'manufactur' is not a valid view function or pattern name.` for a typo or a invalid view name.
`Reverse for 'manufacturer' with no arguments not found. 1 pattern(s) tried: ['dcim/manufacturers/(?P<slug>[-a-zA-Z0-9_]+)/\\Z']` for a missing `kwargs` attribute